uint/src/uint.rs

```diff
@@ -813,6 +813,19 @@ macro_rules! construct_uint {
 				arr
 			}
 		}
+
+		impl From<[u8; $n_words * 8]> for $name {
+			fn from(bytes: [u8; $n_words * 8]) -> Self {
+				bytes[..].as_ref().into()
```

Comment:
just wonder.. Is it possible to convert without copy here?

Reply:
You mean like this?

```rust
		impl From<[u8; $n_words * 8]> for $name {
			fn from(bytes: [u8; $n_words * 8]) -> Self {
				let bytes : [u64; $n_words] = unsafe { ::core::mem::transmute(bytes) };
				$name(bytes)
			}
		}
```

Benchmarking that is no speedup at all though. :/ 

Where is the copy you mention, I fail to see it?

Here's the benchmark I used, maybe it's wrong?

```rust
#[bench]
fn u512_from_fixed_array(b: &mut Bencher) {
    let ary512 : [u8; 64] =  [
        255, 0, 0, 123, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 121, 0, 0, 0, 0,
        0, 213, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 100, 0,
        0, 0, 0, 0, 0, 0, 0, 0,
        0, 45, 0, 0, 67, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 123
    ];
    let ary256 : [u8; 32] =  [
        255, 0, 0, 123, 0, 0, 0, 0,
        0, 0, 0, 0, 0, 0, 0, 0,
        0, 0, 0, 121, 0, 0, 0, 0,
        0, 213, 0, 0, 0, 0, 0, 0,
    ];
    b.iter(|| {
        let n = black_box(1000);
        for _i in 0..n {
            let _ : U512 = black_box(ary512).into();
            let _ : U256 = black_box(ary256).into();
        }
    })
}
```

Reply:
as I understand, in should resolved to this call this: https://github.com/paritytech/parity-common/blob/6d47b026897bc940f97b1cb8ea56d79c3facda2a/uint/src/uint.rs#L777
so..